auth/iam_ipa.go

```diff
@@ -237,17 +240,35 @@ func (ipa *IpaIAMService) login() error {
 	req.Header.Set("referer", fmt.Sprintf("%s/ipa", ipa.host))
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 
-	resp, err := ipa.client.Do(req)
+	var resp *http.Response
+	for i := range requestRetries {
+		resp, err = ipa.client.Do(req)
+		if err == nil {
+			break
+		}
+		// Check for transient network errors
+		if err == io.EOF || strings.Contains(err.Error(), "EOF") ||
+			strings.Contains(err.Error(), "connection reset") ||
+			strings.Contains(err.Error(), "timeout") {
```

Comment:
timeout messages are upper case unless I am looking at the wrong place? e.g.

				err = &timeoutError{err.Error() + " (Client.Timeout exceeded while awaiting headers)"}
				
in https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/net/http/client.go